spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/pulsar/PulsarAutoConfigurationTests.java

```diff
@@ -285,24 +283,29 @@ void injectsExpectedBeans() {
 		}
 
 		@Test
-		@Disabled("Changes in https://github.com/spring-projects/spring-pulsar/issues/593 prevent introspection of the interceptors")
-		void whenHasUseDefinedProducerInterceptorInjectsBean() {
+		<T> void whenHasUseDefinedProducerInterceptorInjectsBean() {
 			ProducerInterceptor interceptor = mock(ProducerInterceptor.class);
 			this.contextRunner.withBean("customProducerInterceptor", ProducerInterceptor.class, () -> interceptor)
-				.run((context) -> assertThat(context).getBean(PulsarTemplate.class)
-					.extracting("interceptorsCustomizers")
-					.asInstanceOf(InstanceOfAssertFactories.LIST)
-					.contains(interceptor));
+				.run((context) -> {
+					PulsarTemplate<?> pulsarTemplate = context.getBean(PulsarTemplate.class);
+					Customizers<ProducerBuilderCustomizer<T>, ProducerBuilder<T>> customizers = Customizers
```

Comment:
@philwebb added this `Customizers` testing utility to ease the burden of verifying customizers. Basically because they are lambdas we have to execute them and see if they "customized" what we expected them to customizer (and in order). In this case, we know our producer builder customizers are going to call `ProducerBuilder.intercept` with the specified interceptor.